crux-http-client/src/crux/remote_api_client.clj

```diff
@@ -91,7 +91,8 @@
                                            (= :stream (:as http-opts)) slurp))]
          (throw (case (::err/error-type error-data)
                   :illegal-argument (err/illegal-arg (::err/error-key error-data) error-data)
-                  :node-out-of-sync (err/node-out-of-sync error-data))))
+                  :node-out-of-sync (err/node-out-of-sync error-data)
+                  (err/illegal-arg "Generic remote client error" error-data))))
```

Comment:
Fair enough to add a default case, although it's not an illegal argument at that point - I'd go with `(ex-info "..." error-data)`.